sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4601,6 +4601,231 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
+      case _ =>
+        DataTypeMismatch(
+          errorSubClass = "ARRAY_FUNCTION_DIFF_TYPES",
+          messageParameters = Map(
+            "functionName" -> toSQLId(prettyName),
+            "dataType" -> toSQLType(ArrayType),
+            "leftType" -> toSQLType(first.dataType),
+            "rightType" -> toSQLType(third.dataType)
+          )
+        )
+    }
+  }
+
+  override def eval(input: InternalRow): Any = {
+    val value1 = first.eval(input)
+    if (value1 != null) {
+      val value2 = second.eval(input)
+      if (value2 != null) {
+        val value3 = third.eval(input)
+        return nullSafeEval(value1, value2, value3)
+      }
+    }
+    null
+  }
+
+  override def nullSafeEval(arr: Any, pos: Any, item: Any): Any = {
+    val baseArr = arr.asInstanceOf[ArrayData]
+    val posInt = pos.asInstanceOf[Int]
+    val arrayElementType = dataType.asInstanceOf[ArrayType].elementType
+
+    val newPosExtendsArrayLeft = posInt < 0 && math.abs(posInt) > baseArr.numElements() - 1
+
+    val itemInsertionIndex = if (newPosExtendsArrayLeft) {
+      0
+    } else if (posInt < 0) {
+      posInt + baseArr.numElements()
+    } else {
+      posInt
+    }
+
+    val newArrayLength = if (newPosExtendsArrayLeft) {
+      math.abs(posInt) + 1
+    } else {
+      math.max(baseArr.numElements() + 1, itemInsertionIndex + 1)
+    }
+
+    val newArray = new Array[Any](newArrayLength)
+
+    baseArr.foreach(arrayElementType, (i, v) => {
+      var elementPosition = i
+      if (newPosExtendsArrayLeft) {
+        elementPosition = elementPosition + 1 + math.abs(posInt + baseArr.numElements())
+      }
+      if (i >= itemInsertionIndex && !newPosExtendsArrayLeft) {
+        elementPosition = elementPosition + 1
+      }
+      newArray(elementPosition) = v
+    })
+
+    newArray(itemInsertionIndex) = item
+
+    return new GenericArrayData(newArray)
+  }
+
+  override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val f = (arrExpr: ExprCode, posExpr: ExprCode, itemExpr: ExprCode) => {
+      val arr = arrExpr.value
+      val pos = posExpr.value
+      val item = itemExpr.value
+
+      val itemInsertionIndex = ctx.freshName("itemInsertionIndex")
+      val newPosExtendsArrayLeft = ctx.freshName("newPosExtendsArrayLeft")
+      val adjustedAllocIdx = ctx.freshName("adjustedAllocIdx")
+      val resLength = ctx.freshName("resLength")
+      val insertedItemIsNull = ctx.freshName("insertedItemIsNull")
+      val i = ctx.freshName("i")
+      val j = ctx.freshName("j")
+      val values = ctx.freshName("values")
+
+      val allocation = CodeGenerator.createArrayData(
+        values, elementType, resLength, s"$prettyName failed.")
+      val assignment = CodeGenerator.createArrayAssignment(values, elementType, arr,
+        adjustedAllocIdx, i, true)
+
+      val defaultIntValue = CodeGenerator.defaultValue(CodeGenerator.JAVA_INT, false)
+      s"""
+         |${CodeGenerator.JAVA_INT} $itemInsertionIndex = 0;
```

Comment:
`CodeGenerator.JAVA_*` only  used here, the following code is no longer used, better unify

Reply:
hmm could you clarify what the replacement should be? I don't quite understand- would you like to replace the '0' with defaultIntValue?

Reply:
for example, line 4730 change to use `int ` or other `int`(such as line 4752) change to use `${CodeGenerator.JAVA_INT}`

Reply:
oh i see- good catch, thank you

Reply:
Amended, thank you for reviewing!